### PR TITLE
ceph-daemon: consolidate NamedTemporaryFile logic

### PR DESCRIPTION
**Pull request by mgfritch**
Adds missing chown of the uid/gid for `ceph-volume` temp files as well as consolidates the logic for creating a `NamedTemporaryFile`.

Signed-off-by: Michael Fritch <mfritch@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/ceph-daemon/ceph-daemon [hunk 1]
```diff
@@ -961,16 +963,11 @@ def command_bootstrap():
...
-    os.fchown(tmp_bootstrap_keyring.fileno(), uid, gid)
-    tmp_bootstrap_keyring.write(keyring)
-    tmp_bootstrap_keyring.flush()
+    tmp_bootstrap_keyring = write_tmp(keyring, uid, gid)
 
     # create initial monmap, tmp monmap file
     logger.info('Creating initial monmap...')
-    tmp_monmap = tempfile.NamedTemporaryFile(mode='w')
-    os.fchmod(tmp_monmap.fileno(), 0o644)
+    tmp_monmap = write_tmp('', uid, gid, mode=0o644)
```
Comment (liewegas) [inline src/ceph-daemon/ceph-daemon]: I think this should work with 0o600 too.. which would let us drop the mode arg complete?

File: src/ceph-daemon/ceph-daemon [hunk 2]
```diff
@@ -1308,6 +1297,7 @@ def command_ceph_volume():
     if args.fsid:
         make_log_dir(args.fsid)
 
+    (uid, gid) = extract_uid_gid()
```
Comment (liewegas) [inline src/ceph-daemon/ceph-daemon]: I think these temp files can be root:root and we can skip extracting the uid/gid from the container

File: src/ceph-daemon/ceph-daemon (hunk 1 above)
Comment (mgfritch) [reply src/ceph-daemon/ceph-daemon]: yup, this is not needed...

File: src/ceph-daemon/ceph-daemon (hunk 2 above)
Comment (mgfritch) [reply src/ceph-daemon/ceph-daemon]: I'm not sure about all `ceph-volume` command, but the basic ones seem aok when using the effective uid/gid ..

File: src/ceph-daemon/ceph-daemon [hunk 3]
```diff
@@ -1342,7 +1342,7 @@ def command_ceph_volume():
     if args.fsid:
         make_log_dir(args.fsid)
 
-    (uid, gid) = extract_uid_gid()
+    (uid, gid) = (os.geteuid(), os.getegid())
```
Comment (liewegas) [inline src/ceph-daemon/ceph-daemon]: actually, root:root.  c-v runs as root in the container.  as long as these files are readable by root all is well.

Comment (mgfritch) [reply src/ceph-daemon/ceph-daemon]: updated to use uid/gid 0 (root:root)

**Pull request merged**